benthomasson/eda/tests/unit/event_source/test_webhook.py

```diff
@@ -0,0 +1,53 @@
+import pytest
+import aiohttp
+import asyncio
+
+from plugins.event_source.webhook import main as webhook_main
+
+
+async def start_server(queue, args):
+    await webhook_main(queue, args)
+
+
+async def post_code(server_task, info):
+    async with aiohttp.ClientSession() as session:
+        async with session.post('http://localhost:8000/payload', json=info) as resp:
+            print(resp.status)
+ 
+    server_task.cancel()
+
+
+async def cancel_code(server_task):
+    server_task.cancel()
+
+
+@pytest.mark.asyncio
+async def test_cancel():
+    queue = asyncio.Queue()
+
+    args = {"host": 'localhost', "port": 8000}
+    plugin_task = asyncio.create_task(start_server(queue, args))
+    asyncio.sleep(1)
+    cancel_task = asyncio.create_task(cancel_code(plugin_task))
+
+    with pytest.raises(asyncio.CancelledError):
+        await asyncio.gather(plugin_task, cancel_task)
```

Comment:
I think you are missing here the parameter `return_exceptions=True`
https://docs.python.org/3/library/asyncio-task.html#asyncio.gather

Reply:
No, I need it to raise the cancelled error, which is the default `return_exceptions=False` means